homeassistant/components/sensor/teleinfo.py

```diff
@@ -0,0 +1,156 @@
+"""
+Support for reading Teleinfo data from a serial port.
+
+Teleinfo is a French specific protocol used in electricity smart meters.
+It provides real time information on power consumption, rates and current on
+a user accessible serial port.
+
+For more details about this platform, please refer to the documentation at
+https://www.enedis.fr/sites/default/files/Enedis-NOI-CPT_02E.pdf
+
+Work based on https://github.com/nlamirault
+
+Sample configuration.yaml
+
+    sensor:
+    - platform: teleinfo
+        name: "EDF teleinfo"
+        serial_port: "/dev/ttyAMA0"
+    - platform: template
+        sensors:
+        teleinfo_base:
+            value_template: '{{ (states.sensor.edf_teleinfo.
+                attributes["BASE"] | float / 1000) | round(0) }}'
+            unit_of_measurement: 'kWh'
+            icon_template: mdi:flash
+    - platform: template
+        sensors:
+        teleinfo_iinst1:
+            value_template: '{{ states.sensor.edf_teleinfo.
+                attributes["IINST1"] | int }}'
+            unit_of_measurement: 'A'
+            icon_template: mdi:flash
+
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, EVENT_HOMEASSISTANT_STOP, ATTR_ATTRIBUTION)
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['pyserial-asyncio==0.4']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_SERIAL_PORT = 'serial_port'
+
+CONF_ATTRIBUTION = "Provided by EDF Teleinfo."
+
+DEFAULT_NAME = "Serial Teleinfo Sensor"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_SERIAL_PORT): cv.string,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the Serial sensor platform."""
+    name = config.get(CONF_NAME)
+    port = config.get(CONF_SERIAL_PORT)
+    sensor = SerialTeleinfoSensor(name, port)
+
+    hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_STOP, sensor.stop_serial_read())
+    async_add_entities([sensor], True)
```

Comment:
No need for `True` since you aren't defining an `update` method.

Reply:
Ok